community/neo4j/src/test/java/ConcurrentChangesOnEntitiesTest.java

```diff
@@ -0,0 +1,240 @@
+                {
+                    Node n = db.getNodeById( nodeId );
+                    n.addLabel( DynamicLabel.label( "A" ) );
+                    tx.success();
+                }
+                catch ( Exception e )
+                {
+                    ex.set( e );
+                }
+            }
+        } );
+
+        t1.start();
+        t2.start();
+
+        t1.join();
+        t2.join();
+
+        if ( ex.get() != null )
+        {
+            throw ex.get();
+        }
+
+        db.shutdown();
+
+        LogProvider logProvider = FormattedLogProvider.toOutputStream( System.out );
+        try
+        {
+            ConsistencyCheckService.Result result = new ConsistencyCheckService().runFullConsistencyCheck(
+                    testDirectory.graphDbDir(), new Config(), ProgressMonitorFactory.textual( System.err ),
+                    logProvider );
+            assertTrue( result.isSuccessful() );
+        }
+        catch ( ConsistencyCheckIncompleteException e )
+        {
+            throw new RuntimeException( e );
+        }
+    }
+
+    @Test
+    public void setConcurrentlySamePropertyWithDifferentValuesOnANode() throws Throwable
+    {
+
+        final GraphDatabaseService db = new GraphDatabaseFactory().newEmbeddedDatabase( testDirectory.graphDbDir() );
+
+        final long nodeId = initWithNode( db );
+
+        Thread t1 = new Thread( new Runnable()
+        {
+            @Override
+            public void run()
+            {
+                try ( Transaction tx = db.beginTx() )
+                {
+                    Node n = db.getNodeById( nodeId );
+                    n.setProperty( "a", 0.788 );
+                    tx.success();
+
+                }
+            }
+        } );
+
+        Thread t2 = new Thread( new Runnable()
+        {
+            @Override
+            public void run()
+            {
+                try ( Transaction tx = db.beginTx() )
+                {
+                    Node n = db.getNodeById( nodeId );
+                    n.setProperty( "a", new double[]{0.999, 0.77} );
+                    tx.success();
+                }
+            }
+        } );
+
+        t1.start();
+        t2.start();
+
+        t1.join();
+        t2.join();
+
+        db.shutdown();
+
+        LogProvider logProvider = FormattedLogProvider.toOutputStream( System.out );
+        try
+        {
+            ConsistencyCheckService.Result result = new ConsistencyCheckService().runFullConsistencyCheck(
+                    testDirectory.graphDbDir(), new Config(), ProgressMonitorFactory.textual( System.err ),
+                    logProvider );
+            assertTrue( result.isSuccessful() );
+        }
+        catch ( ConsistencyCheckIncompleteException e )
+        {
+            throw new RuntimeException( e );
+        }
+    }
+
+    @Test
+    public void setConcurrentlySamePropertyWithDifferentValuesOnARelationship() throws Throwable
```

Comment:
Since there is nothing coordinating the two threads, there is no guarantee that the issue will manifest.